integration/run_test.go

```diff
@@ -465,3 +466,181 @@ func TestKritisPods(t *testing.T) {
+				cmd.Dir = "../"
+				output, err := integration_util.RunCmdOut(cmd)
+				if err != nil {
+					t.Fatalf("kubectl delete failed: %s %v", output, err)
+				}
+			},
+		},
+	}
+
+	ns, deleteNs := setupNamespace(t)
+	defer deleteNs()
+	createGACSecret(t, ns)
+	deleteKritis := initKritis(t, ns)
+	defer deleteKritis()
+	if err := kubernetesutil.WaitForDeploymentToStabilize(client, ns.Name,
+		fmt.Sprintf("kritis-validation-hook-%s", ns.Name), 2*time.Minute); err != nil {
+		t.Fatalf("Timed out waiting for deployment to stabilize")
+	}
+	createCRDExamples(t, ns)
+	waitForCRDExamples(t, ns)
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			defer testCase.cleanup(t)
+
+			cmd := exec.Command(testCase.args[0], testCase.args[1:]...)
+			cmd.Dir = testCase.dir
+			output, err := integration_util.RunCmdOut(cmd)
+			if err != nil {
+				if !testCase.shouldSucceed {
+					return
+				}
+				t.Fatalf("testCase cmd failed: %s %v\n%s", output,
+					err, getKritisLogs(t))
+
+			}
+			if !testCase.shouldSucceed {
+				t.Fatalf("deployment should have failed but succeeded\n%s",
+					getKritisLogs(t))
+			}
+
+			for _, p := range testCase.pods {
+				if err := kubernetesutil.WaitForPodReady(client.CoreV1().Pods(ns.Name), p.name); err != nil {
+					t.Fatalf("Timed out waiting for pod ready\n%s\n%s",
+						getKritisLogs(t),
+						output)
+				}
+			}
+
+			for _, d := range testCase.deployments {
+				if err := kubernetesutil.WaitForDeploymentToStabilize(client, ns.Name, d.name, 10*time.Minute); err != nil {
+					t.Fatalf("Timed out waiting for deployment to stabilize\n%s",
+						getKritisLogs(t))
+				}
+				if testCase.deploymentValidation != nil {
+					deployment, err := client.AppsV1().Deployments(ns.Name).Get(d.name, meta_v1.GetOptions{})
+					if err != nil {
+						t.Fatalf("Could not find deployment: %s %s\n%s", ns.Name, d, getKritisLogs(t))
+					}
+					testCase.deploymentValidation(t, deployment)
+				}
+			}
+			// get
+			cmd = exec.Command("kubectl", "get", "po")
+			cmd.Dir = "../"
+			pods, err := integration_util.RunCmdOut(cmd)
+			if err != nil {
+				t.Fatalf("kubectl get for kritis-server failed: %s %v", output, err)
+			}
+			logrus.Infof("kubectl get po output: %s", string(pods))
+
+			cmd = exec.Command("kubectl", "get", "po",
+				"-l", "label=kritis-validation-hook",
+				"-o", "custom-columns=:metadata.name",
+				"--no-headers=true")
+			cmd.Dir = "../"
+			kritisPod, err := integration_util.RunCmdOut(cmd)
+			if err != nil {
+				t.Fatalf("kubectl get for kritis-server failed: %s %v", output, err)
+			}
+			logrus.Infof("kritisPod output: %s", string(kritisPod))
+
+			// as kubectl exec opens a tty, this command only wait for it to finish
+			cmd = exec.Command("kubectl", "exec", strings.TrimSpace(string(kritisPod)),
+				"--",
+				"/kritis/kritis-server", "--run-cron")
+			err = cmd.Start()
+			if err != nil {
+				logrus.Infof("Error starting Cmd: %v", err)
```

Comment:
Including cmd here might be helpful. Also, this should be t.Fatalf instead of Info+os.Exit.